src/lib/Zikula/Common/Translator/Translator.php

```diff
@@ -0,0 +1,262 @@
+ *
+ * This work is contributed to the Zikula Foundation under one or more
+ * Contributor Agreements and licensed to You under the following license:
+ *
+ * @license    GNU/LGPLv3 (or at your option, any later version).
+ * @package    Zikula
+ * @subpackage Zikula_Translate
+ *
+ * Please see the NOTICE file distributed with this source code for further
+ * information regarding copyright and licensing.
+ */
+namespace Zikula\Common\Translator;
+
+use Symfony\Component\HttpKernel\CacheWarmer\WarmableInterface;
+use Symfony\Component\Translation\Translator as BaseTranslator;
+use Symfony\Component\Translation\MessageSelector;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Translator.
+ */
+class Translator extends BaseTranslator implements WarmableInterface {
+	protected $container;
+	protected $domain;
+	protected $loaderIds;
+	protected $options = array (
+			'cache_dir' => null,
+			'debug' => false,
+			'resource_files' => array () 
+	);
+	
+	/**
+	 *
+	 * @var array
+	 */
+	private $resourceLocales;
+	/**
+	 * Constructor.
+	 *
+	 * Available options:
+	 *
+	 * * cache_dir: The cache directory (or null to disable caching)
+	 * * debug: Whether to enable debugging or not (false by default)
+	 * * resource_files: List of translation resources available grouped by locale.
+	 *
+	 * @param ContainerInterface $container
+	 *        	A ContainerInterface instance
+	 * @param MessageSelector $selector
+	 *        	The message selector for pluralization
+	 * @param array $loaderIds
+	 *        	An array of loader Ids
+	 * @param array $options
+	 *        	An array of options
+	 *        	
+	 * @throws \InvalidArgumentException
+	 */
+	public function __construct(ContainerInterface $container, MessageSelector $selector, $loaderIds = array(), array $options = array()) {
+		$this->container = $container;
+		$this->selector = $selector ?: new MessageSelector();
+		$this->loaderIds = $loaderIds;
+		// check option names
+		if ($diff = array_diff ( array_keys ( $options ), array_keys ( $this->options ) )) {
+			throw new \InvalidArgumentException ( sprintf ( 'The Translator does not support the following options: \'%s\'.', implode ( '\', \'', $diff ) ) );
+		}
+		
+		$this->domain = 'zikula';
+		
+		$this->options = array_merge ( $this->options, $options );
+		$this->resourceLocales = array_keys ( $this->options ['resource_files'] );
+		if (null !== $this->options ['cache_dir'] && $this->options ['debug']) {
+			$this->loadResources ();
+		}
+		parent::__construct ( $container->getParameter ( 'kernel.default_locale' ), $selector, $this->options ['cache_dir'], $this->options ['debug'] );
+	}
+	/**
+	 *
+	 * @ERROR!!!
+	 *
+	 */
+	public function warmUp($cacheDir) {
+		// skip warmUp when translator doesn't use cache
+		if (null === $this->options ['cache_dir']) {
+			return;
+		}
+		foreach ( $this->resourceLocales as $locale ) {
+			$this->loadCatalogue ( $locale );
+		}
+	}
+	/**
+	 *
+	 * @ERROR!!!
```

Comment:
Well from first look I can tell only that this might get overridden by Eclipse. Original was {@inheritdoc} I have copied this first from https://github.com/symfony/symfony/blob/2.8/src/Symfony/Bundle/FrameworkBundle/Translation/Translator.php
to my dev setup and during developement and test this was probably altered. I have mentioned in Todo section cleaning + doc blocks so this will be checked and fixed.